server.go

```diff
@@ -305,21 +305,23 @@ func (s *Server) serveConnection(ctx context.Context, c *serverConn, config serv
 }
 
 func (s *Server) serveRequest(res *responseWriter, req *Request) (err error) {
-	args := req.Args
+	cmds := req.Cmds
 
-	switch req.Cmd {
-	case "PING":
-		msg := "PONG"
-		req.ParseArgs(&msg)
-		res.Write(msg)
+	for _, cmd := range cmds {
+		switch cmd.Cmd {
+		case "PING":
+			msg := "PONG"
+			req.ParseArgs(&msg)
+			res.Write(msg)
 
-	default:
-		err = s.serveRedis(res, req)
-		args.Close()
-	}
+		default:
+			err = s.serveRedis(res, req)
```

Comment:
This is weird isn't it? It's gonna call `serveRedis` multiple times for the same request if there are many arguments.